utils/subprocess/executor.go

```diff
@@ -192,12 +196,24 @@ func (s *Subprocess) IsOn() bool {
 	return s.isRunning.Load() && s.processMonitoring.IsOn()
 }
 
-// Wait waits for the command to exit and waits for any copying to
-// stdin or copying from stdout or stderr to complete.
-//
-// The command must have been started by Start.
-func (s *Subprocess) Wait() error {
-	return s.command.cmdWrapper.cmd.Wait()
+// Wait waits for the command to stop existing on the system.
+// This allows check to work if the underlying process was stopped.
+func (s *Subprocess) Wait(ctx context.Context) (err error) {
+	pid := s.command.cmdWrapper.cmd.Process.Pid
```

Comment:
process could be nil so do a similar check to https://cs.opensource.google/go/go/+/refs/tags/go1.24.4:src/os/exec/exec.go;drc=7e0950861703c93ea3e7f122c1c594851919261e;l=916